syntax = "proto3";
package dbservice;




import "database_delivery_sdk/model/database_delivery/dbinstance.proto";

/*
ListDBService请求
*/
message ListDBServiceRequest {
    /*
    页码
    */
    int32 page = 1 ;
    /*
    每页数据量
    */
    int32 pageSize = 2 ;
}

/*
ListDBService返回
*/
message ListDBServiceResponse {
message List {
message Owner {
    /*
    用户ID
    */
    string instanceId = 1 ;
    /*
    邮箱
    */
    string user_email = 2 ;
    /*
    电话号码
    */
    string user_tel = 3 ;
    /*
    用户类型
    */
    string user_type = 4 ;
    /*
    名称
    */
    string nickname = 5 ;
}
        
    
    /*
    数据库实例列表
    */
    repeated database_delivery.DBInstance dbInstance = 1 ;
    /*
    运维负责人列表
    */
    repeated Owner owner = 2 ;
    /*
    数据库服务ID
    */
    string instanceId = 3 ;
    /*
    名称
    */
    string name = 4 ;
    /*
    数据库类型
    */
    string dbType = 5 ;
    /*
    描述
    */
    string desc = 6 ;
    /*
    变更时间
    */
    int64 updatedTime = 7 ;
}
        
    
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
ListDBServiceApi返回
*/
message ListDBServiceResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    dbservice.ListDBServiceResponse data = 4 ;
}

